File: app/src/main/java/com/ramez/shopp/adapter/ProductOptionAdapter.java
package com.ramez.shopp.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.ramez.shopp.CallBack.DataCallback;
import com.ramez.shopp.Models.ProductBarcode;
import com.ramez.shopp.R;
import com.ramez.shopp.databinding.RowProductOptionBinding;

import java.util.List;

public class ProductOptionAdapter extends RecyclerView.Adapter<ProductOptionAdapter.MyHolder> {

    public List<ProductBarcode> productBarcodes;
    public Context context;
    public LayoutInflater inflater;
    public int selectedIndex = -1;
    DataCallback dataCallback;
    private int lastIndex = 0;


    public ProductOptionAdapter(Context context, List<ProductBarcode> productBarcodes, DataCallback dataCallback) {
        this.productBarcodes = productBarcodes;
        this.context = context;
        this.inflater = LayoutInflater.from(context);
        this.dataCallback = dataCallback;

    }

    @Override
    public MyHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        RowProductOptionBinding view = RowProductOptionBinding.inflate(LayoutInflater.from(context), parent, false);
        MyHolder holder = new MyHolder(view);
        return holder;
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(final MyHolder holder, final int position) {

        ProductBarcode productOptionModel = productBarcodes.get(position);
        String wightName = productOptionModel.getProductUnits().getUnitName();
        //NumberHandler.formatDouble(productOptionModel.getWeight(), 0)
        //                + " " +
        holder.binding.btnCategory.setText(wightName);


        if (lastIndex == position) {

            holder.binding.btnCategory.setBackground(ContextCompat.getDrawable(context, R.drawable.round_corner_green));
            holder.binding.btnCategory.setTextColor(ContextCompat.getColor(context, R.color.green3));


        } else {
            holder.binding.btnCategory.setBackground(ContextCompat.getDrawable(context, R.drawable.round_corner_grey));
            holder.binding.btnCategory.setTextColor(ContextCompat.getColor(context, R.color.black));


        }


    }


    @Override
    public int getItemCount() {
        return productBarcodes.size();
    }

    class MyHolder extends RecyclerView.ViewHolder {
        RowProductOptionBinding binding;

        public MyHolder(RowProductOptionBinding itemView) {
            super(itemView.getRoot());
            binding = itemView;
            binding.btnCategory.setOnClickListener(v -> {
                int position = getBindingAdapterPosition();
                selectedIndex = position;
                lastIndex = position;
                ProductBarcode optionModel = productBarcodes.get(position);
                Log.i("tag", "Log 1" + optionModel.getId());
                notifyDataSetChanged();
                if (dataCallback != null) {
                    dataCallback.dataResult(optionModel, "success", true);
                }
            });


        }


    }


}

File: app/src/main/java/com/ramez/shopp/adapter/ProductOptionAdapter.kt
package com.ramez.shopp.adapter


import android.annotation.SuppressLint
import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.ramez.shopp.CallBack.DataCallback
import com.ramez.shopp.Models.ProductBarcode
import com.ramez.shopp.R
import com.ramez.shopp.databinding.RowProductOptionBinding


class ProductOptionAdapter(
    var context: Context,
    var productBarcodes: List<ProductBarcode>,
    var dataCallback: DataCallback?
) :
        RecyclerView.Adapter<ProductOptionAdapter.MyHolder>() {
    var inflater: LayoutInflater = LayoutInflater.from(context)
    var selectedIndex = -1
    private var lastIndex = 0
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyHolder {
        val view =
            RowProductOptionBinding.inflate(LayoutInflater.from(context), parent, false)
        return MyHolder(view)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: MyHolder, position: Int) {
        val productOptionModel = productBarcodes[position]
        val wightName = productOptionModel.productUnits?.unitName?:""
        holder.binding.btnCategory.text = wightName
        if (lastIndex == position) {
            holder.binding.btnCategory.background =
                ContextCompat.getDrawable(context, R.drawable.round_corner_green)
            holder.binding.btnCategory.setTextColor(ContextCompat.getColor(context, R.color.green3))
        } else {
            holder.binding.btnCategory.background =
                ContextCompat.getDrawable(context, R.drawable.round_corner_grey)
            holder.binding.btnCategory.setTextColor(ContextCompat.getColor(context, R.color.black))
        }
    }

    override fun getItemCount(): Int {
        return productBarcodes.size
    }

    @SuppressLint("NotifyDataSetChanged")
    inner class MyHolder(var binding: RowProductOptionBinding) : RecyclerView.ViewHolder(
        binding.root
    ) {
        init {
            binding.btnCategory.setOnClickListener { v ->
                val position = bindingAdapterPosition
                selectedIndex = position
                lastIndex = position
                if (position >= 0 && position < productBarcodes.size) {
                    val productBarcode = productBarcodes[position]
                    Log.i("tag", "Log 1" + productBarcode.id)
                    notifyDataSetChanged()
                    if (dataCallback != null) {
                        dataCallback?.dataResult(productBarcode, "success", true)
                    }
                }
            }
        }
    }

}

